plugins/module_utils/ndi.py

```diff
@@ -0,0 +1,444 @@
+
+    def query_entry(self, ig_name, site_name, epoch_delta_job_id):
+        epoch_delta_ig_path = self.epoch_delta_ig_path.format(ig_name, site_name, epoch_delta_job_id)
+        path = "{0}/individualTable?epochStatus=BOTH_EPOCHS".format(epoch_delta_ig_path)
+        obj = self.nd.query_obj(path, prefix = self.prefix)
+        return obj['entries']
+
+    def format_event_severity(self, events_severity):
+        result = {}
+        for each in events_severity:
+            event_severity_type = each.get("bucket").lower().split('_')[-1]
+            result[event_severity_type] = {}
+            for output in each.get("output"):
+                epoch = output.get("bucket").lower()
+                epoch_count = output.get("count")
+                result[event_severity_type][epoch] = epoch_count
+        return result
+
+    def format_impacted_resource(self, impacted_resource):
+        result = {}
+        for each in impacted_resource:
+            resource = each.get("bucket").lower()
+            result[resource] = {}
+            for output in each.get("output"):
+                for epoch in output.get("output"):
+                    epoch_type = epoch.get("bucket").lower()
+                    count = epoch.get("count")
+                    result[resource][epoch_type] = count
+        return result
+
+    def query_event_severity(self, ig_name, site_name, epoch_delta_job_id):
+        epoch_delta_ig_path = self.epoch_delta_ig_path.format(ig_name, site_name, epoch_delta_job_id)
+        path = "{0}/eventSeverity".format(epoch_delta_ig_path)
+        event_severity = self.query_data(path)
+        formated_event_severity = self.format_event_severity(event_severity)
+        return formated_event_severity
+
+    def query_impacted_resource(self, ig_name, site_name, epoch_delta_job_id):
+        epoch_delta_ig_path = self.epoch_delta_ig_path.format(ig_name, site_name, epoch_delta_job_id)
+        path = "{0}/impactedResource".format(epoch_delta_ig_path)
+        impacted_resource = self.query_data(path)
+        formated_impacted_resource = self.format_impacted_resource(impacted_resource)
+        return formated_impacted_resource
+
+    def format_messages(self, messages):
+        result = {}
+        for message in messages:
+            msg = message.get("message")
+            severity = message.get("severity").lower()
+            result[severity] = msg
+        return result
+
+    def query_messages(self, path):
+        obj = self.nd.query_obj(path, prefix=self.prefix)
+        if obj.get("messages") is not None:
+            result = self.format_messages(obj.get("messages"))
+            return result
+
+    def query_compliance_smart_event(self, ig_name, site_name, compliance_epoch_id):
+        ig_base_path = self.event_insight_group_path.format(ig_name, site_name)
+        path = "{0}/smartEvents?%24epochId={1}&%24page=0&%24size=10&%24sort=-severity&category=COMPLIANCE".format(ig_base_path, compliance_epoch_id)
```

Comment:
how do we deal with this if there are more than 10 elements in the compliance smart event list?

Reply:
Not addressed yet.